### PR TITLE
Use utils to define config

### PR DESCRIPTION
**Pull request by thetutlage**
Updated the config file to use the new utils. 

Also, I have made a major change to the way `STORE_CORS` or `ADMIN_CORS` are computed. I using the environment variable directly and not appending Medusa URLS to them. So please let me know if we still need Medusa URLs like `https://docs.medusajs.com` in CORS config

### CODE REVIEW

File: medusa-config.js [hunk 1]
```diff
@@ -1,133 +1,14 @@
...
-    break;
-  case "test":
-    ENV_FILE_NAME = ".env.test";
-    break;
-  case "development":
-  default:
-    ENV_FILE_NAME = ".env";
-    break;
-}
+loadEnv(process.env.NODE_ENV, process.cwd())
```
Comment (shahednasser) [inline medusa-config.js:3]: Did we agree on making these optional? Or would this be a change made later?

File: medusa-config.js [hunk 2]
```diff
@@ -1,133 +1,14 @@
...
-  databaseUrl: DATABASE_URL,
+module.exports = defineConfig({
+  databaseUrl: process.env.DATABASE_URL,
   http: {
-    storeCors: STORE_CORS,
-    adminCors: ADMIN_CORS,
-    authCors: process.env.AUTH_CORS || ADMIN_CORS,
+    storeCors: process.env.STORE_CORS,
+    adminCors: process.env.ADMIN_CORS,
+    authCors: process.env.AUTH_CORS,
```
Comment (shahednasser) [inline medusa-config.js:10]: Should we keep default values for these? No strong opinions here just wanna hear some thoughts

**Review by shahednasser — APPROVED** [review]
> So please let me know if we still need Medusa URLs like https://docs.medusajs.com in CORS config

No i think it's fine, we moved those to the `create-medusa-app` to set now.

File: medusa-config.js (hunk 1 above)
Comment (thetutlage) [reply medusa-config.js:3]: Yeah, going to create a PR for it separately.

File: medusa-config.js (hunk 2 above)
Comment (thetutlage) [reply medusa-config.js:10]: Not sure. Don't have any strong opinions either. If others think it should be defined here, I can update the code

**Review by adrien2p — APPROVED** [review]
So much cleaner :D

File: medusa-config.js (hunk 2 above)
Comment (olivermrbl) [reply medusa-config.js:10]: Do we have default values for these somewhere else, if not here? We will definitely need to have the CORS configured by default, but whether they come from a prepopulated `.env` or in-line here is not super important imo

Comment (adrien2p) [reply medusa-config.js:10]: I think they are in the env file in theory no? at least they could be in the env template and env file

Comment (adrien2p) [reply medusa-config.js:10]: Screenshot 2024-05-30 at 09 48 58

Comment (adrien2p) [reply medusa-config.js:10]: also, should we merge postgres_url (modules) and database_url?

Comment (olivermrbl) [reply medusa-config.js:10]: > I think they are in the env file in theory no?

If we can guarantee that they will always be in the `.env` for new projects, then I am fine keeping it as is. 

> also, should we merge postgres_url (modules) and database_url?

Think that would be nice!

Comment (thetutlage) [reply medusa-config.js:10]: Okay, let's leave it like this for now and see if someone gets confused. We can always add them back.

**Pull request merged**